#! /bin/sh /usr/share/dpatch/dpatch-run
## 31_hauppauge_hvr_1100.dpatch by Mario Limonciello <superm1@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add support for the Hauppauge HVR-1100 Remote (LP: #203493)

@DPATCH@
diff -urNad lirc-0.8.3~pre1~/doc/lirc.hwdb lirc-0.8.3~pre1/doc/lirc.hwdb
--- lirc-0.8.3~pre1~/doc/lirc.hwdb	2008-03-24 00:50:43.000000000 -0500
+++ lirc-0.8.3~pre1/doc/lirc.hwdb	2008-03-24 00:51:24.000000000 -0500
@@ -74,6 +74,7 @@
 Freecom DVB-T USB Stick;devinput;none;hw_default;freecom/lircd.conf.freeconf
 Hauppauge TV card;hauppauge;lirc_dev lirc_i2c;hw_default;hauppauge/lircd.conf.hauppauge;
 Hauppauge Nova-T 500;devinput;none;hw_devinput;hauppauge/lircd.conf.hauppauge_novat500;
+Hauppauge HVR-1100;hauppauge;devinput;hw_devinput;hauppauge/lircd.conf.hauppauge;
 Hauppauge HVR-1300;hauppauge;lirc_dev lirc_i2c;hw_default;hauppauge/lircd.conf.hauppauge;
 Hauppauge DVB-s card (ver. 2.1);hauppauge_dvb;lirc_dev;hw_default;hauppauge/lircd.conf.hauppauge;
 Hercules Smart TV Stereo (card 100);hercules_smarttv_stereo;lirc_dev lirc_i2c;hw_default;hercules/lircd.conf.smarttv_stereo;
diff -urNad lirc-0.8.3~pre1~/remotes/hauppauge/lircd.conf.hauppauge lirc-0.8.3~pre1/remotes/hauppauge/lircd.conf.hauppauge
--- lirc-0.8.3~pre1~/remotes/hauppauge/lircd.conf.hauppauge	2005-10-05 15:28:15.000000000 -0500
+++ lirc-0.8.3~pre1/remotes/hauppauge/lircd.conf.hauppauge	2008-03-24 00:50:54.000000000 -0500
@@ -272,3 +272,64 @@
 end remote
 
 
+begin remote
+  name            HVR-1100
+  bits            16
+  eps             30
+  aeps            100
+  one             0     0
+  zero            0     0
+  gap             135862
+  pre_data_bits   16
+  pre_data        0x8001
+
+  begin codes
+       Go              0x0161
+       power           0x0074
+       Tv              0x0179
+       Videos          0x0189
+       Music           0x0188
+       Pictures        0x016F
+       Guide           0x016D
+       Radio           0x0181
+       Up              0x0067
+       Down            0x006C
+       Left            0x0069
+       Right           0x006A
+       Ok              0x001C
+       Back/edit       0x00AE
+       Menu            0x008B
+       Vol+            0x0073
+       Vol-            0x0072
+       PrevChan        0x019C
+       Chan+           0x0192
+       Chan-           0x0193
+       Mute            0x0071
+       Record          0x00A7
+       Stop            0x0080
+       replay          0x00A8
+       SKip            0x00D0
+       play            0x00CF
+       Previous        0x00A5
+       Next            0x00A3
+       Pause           0x0077
+       1               0x0002
+       2               0x0003
+       3               0x0004
+       4               0x0005
+       5               0x0006
+       6               0x0007
+       7               0x0008
+       8               0x0009
+       9               0x000A
+       *               0x0184
+       0               0x000B
+       #               0x0172
+       red             0x018E
+       green           0x018F
+       yellow          0x0190
+       blue            0x0191
+
+  end codes
+
+end remote 
